src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp

```diff
@@ -558,10 +557,15 @@ void ArrowColumnToCHColumn::arrowColumnsToCHChunk(Chunk & res, NameToColumnPtr &
                 read_from_nested = nested_tables[nested_table_name]->has(header_column.name);
             }
 
-
-            // TODO: What if some columns were not presented? Insert NULLs? What if a column is not nullable?
             if (!read_from_nested)
-                throw Exception{ErrorCodes::THERE_IS_NO_COLUMN, "Column '{}' is not presented in input data.", header_column.name};
+            {
+                ColumnWithTypeAndName column;
+                column.name = header_column.name;
+                column.type = header_column.type;
+                column.column = header_column.column->cloneResized(num_rows);
+                columns_list.push_back(std::move(column.column));
+                continue;
+            }
```

Comment:
This should be under some special setting (separate setting for ORC/Parquet/Arrow or one for all these formats). And if we want to fill not presented column and the setting `input_format_defaults_for_omitted_fields` is enabled, we should calculate default expression for this column (to do this we should implement `getMissingValues` from IInputFormat for ORC/Parquet/Arrow formats)

Reply:
And `num_rows` could be 0 here

Reply:
Done.